src/zrc/api/validators.py

```diff
@@ -168,6 +169,31 @@ def __call__(self, attrs):
             raise serializers.ValidationError(self.message, code=self.code)
 
 
+class ZaakArchiefStatusValidator:
+    code = "zaak-archiefstatus-invalid"
+    message = _(
+        f"De huidige zaak archiefstatus staat dit request niet toe. Archiefstatus moet gelijk zijn aan {Archiefstatus.nog_te_archiveren}"
+    )
+
+    def set_context(self, serializer):
+        """
+        This hook is called by the serializer instance,
+        prior to the validation call being made.
+        """
+        # Determine the existing instance, if this is an update operation.
+        self.instance = getattr(serializer, "instance", None)
+
+    def __call__(self, attrs):
+        if self.instance:
+            if Archiefstatus.nog_te_archiveren != self.instance.zaak.archiefstatus:
```

Comment:
Ik zou deze `if` en degene hieronder omdraaien:
```suggestion
            if self.instance.zaak.archiefstatus != Archiefstatus.nog_te_archiveren:
```
Op dit moment leest het een beetje apart.